here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/ops/PropertyUtil.java

```diff
@@ -0,0 +1,220 @@
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.app.service.http.ops;
+
+import static com.here.naksha.lib.core.models.payload.events.QueryDelimiter.*;
+import static com.here.naksha.lib.core.models.payload.events.QueryDelimiter.COMMA;
+import static com.here.naksha.lib.core.models.payload.events.QueryOperation.*;
+import static com.here.naksha.lib.core.util.storage.RequestHelper.pRefFromPropPath;
+
+import com.here.naksha.lib.core.exceptions.XyzErrorException;
+import com.here.naksha.lib.core.models.XyzError;
+import com.here.naksha.lib.core.models.payload.events.QueryDelimiter;
+import com.here.naksha.lib.core.models.payload.events.QueryOperation;
+import com.here.naksha.lib.core.models.payload.events.QueryParameter;
+import com.here.naksha.lib.core.models.payload.events.QueryParameterList;
+import com.here.naksha.lib.core.models.storage.*;
+import com.here.naksha.lib.core.util.ValueList;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class PropertyUtil {
+
+  private static final String NULL_PROP_VALUE = ".null";
+
+  /**
+   * Function builds Property Operation (POp) based on property key:value pairs supplied as API query parameter.
+   * We iterate through all the parameters, exclude the keys that match with provided excludeKeys,
+   * and interpret the others by identifying the desired operation.
+   * <p>
+   * Multiple parameter keys result into AND list.
+   * <br>
+   * So, "p.prop_1=value_1&p.prop_2=value_2" will form AND condition as (p.prop_1=value_1 AND p.prop_2=value_2).
+   * </p>
+   *
+   * <p>
+   * Multiple parameter values concatenated with "," (COMMA) delimiter, will result into OR list.
+   * <br>
+   * So, "p.prop_1=value_1,value_11" will form OR condition as (p.prop_1=value_1 OR p.prop_1=value_11).
+   * </p>
+   *
+   * @param queryParams API query parameter from where property search params need to be extracted
+   * @param excludeKeys List of param keys to be excluded (i.e. not part of property search)
+   * @return POp property operation that can be used as part of {@link ReadRequest}
+   */
+  public static @Nullable POp buildOperationForPropertySearchParams(
+      final @Nullable QueryParameterList queryParams, final @Nullable List<String> excludeKeys) {
```

Comment:
nth - minor nhancement: I think `excludeKeys` could be a `Set`, since it's used only for lookups